strawberryfields/io.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Blackbird I/O
+=============
+
+This module contains functions for loading and saving Strawberry
+Fields :class:`~.Program` objects from/to Blackbird scripts.
+
+Summary
+-------
+
+.. autosummary::
+    to_blackbird
+    to_program
+    save
+    load
+
+Code details
+~~~~~~~~~~~~
+"""
+# pylint: disable=protected-access,too-many-nested-blocks
+import os
+import blackbird
+
+from . import ops
+from .program import Program
+
+
+def to_blackbird(prog, version="1.0"):
+    """Convert a Strawberry Fields Program to a Blackbird Program.
+
+    Args:
+        prog (Program): the Strawberry Fields program
+        version (str): Blackbird script version number
+
+    Returns:
+        blackbird.BlackbirdProgram:
+    """
+    bb = blackbird.BlackbirdProgram(name=prog.name, version=version)
+
+    if prog.backend is not None:
+        # set the target
+        bb._target["name"] = prog.backend
+
+    # fill in the quantum circuit
+    for cmd in prog.circuit:
+        op = {"kwargs": {}, "args": []}
+
+        op["op"] = cmd.op.__class__.__name__
+        op["modes"] = [i.ind for i in cmd.reg]
+
+        if "Measure" in op["op"]:
+            # special case to take into account 'select' keyword argument
+            if cmd.op.select is not None:
+                op["kwargs"]["select"] = cmd.op.select
+
+            if cmd.op.p:
+                # argument is quadrature phase
+                op["kwargs"]["phi"] = cmd.op.p[0].x
+
+        else:
+            if cmd.op.p is not None:
```

Comment:
When looking at this piece of the code, it's hard to know what the attribute `p` is. Should we refactor to give a better name (`params`?)

Reply:
I definitely think so. Even `p.x` is not very explanatory, `p.val` would be better.

However, it's still worth thinking about the future of the parameter class - do we want to replace it with something like SymPy, which is now easy to pip install, and potentially could do what the existing parameter class does and more.

Reply:
Let's make an issue to refactor the names `p` and `x` to make them more clear.

The other question about the future of the Parameter class and sympy seems unrelated to the original comment and PR, so I won't answer here.

Reply:
:+1: